<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="cfg_file" value="$(find hexapod_description)/config/cv_config.yaml"/>
    <arg name="rviz" default="false"/>

    <!-- launch gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="worlds/empty.world"/>
        <arg name="paused" value="true"/>
    </include>

    <!-- spawn hexapid -->
    <include file="$(find hexapod_description)/launch/spawn.launch"/>
    <node name="walker" pkg="hexapod_description" type="walker.py" />

    <!-- opencv detector -->
    <node name="detector" pkg="mono_object_finder" type="detector.py"
        args="--config $(arg cfg_file) --gui base" output="screen">
        <remap from="/image" to="/hexapod/camera1/image_raw"/>
        <remap from="/camera_info" to="/hexapod/camera1/camera_info"/>
    </node>

    <!-- Walk to target -->
    <node name="controller" pkg="hexapod_description" type="opencv_walker.py"/>

    <!-- spawn colored cube -->
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model"
        args="-file $(find hexapod_description)/urdf/target.urdf -urdf -model target -z 0.5 -x 1 -y 0" />

    <!-- rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find hexapod_description)/config/config_gazebo.rviz" if="$(arg rviz)"/>

</launch>